r/R/schema.R

```diff
@@ -50,6 +50,34 @@
 #' - `$metadata`: returns the key-value metadata as a named list.
 #'    Modify or replace by assigning in (`sch$metadata <- new_metadata`).
 #'    All list elements are coerced to string.
+#'
+#' @section Metadata:
+#'
+#'   Attributes from the `data.frame` are saved alongside tables so that the
+#'   object can be reconstructed faithfully in R (e.g. with `as.data.frame()`).
+#'   This metadata can be both at the top-level of the `data.frame` (e.g.
+#'   `attributes(df)`) or at the column (e.g. `attributes(df$col_a)`) or element
+#'   level (e.g. `attributes(df[1, "col_a"])`). For example, this allows for
+#'   storing `haven` columns in a table and being able to faithfully re-create
+#'   them when pulled back into R. This metadata is separate from the schema
+#'   (e.g. types of the columns) which is compatible with other Arrow clients.
+#'   The R metadata is only read by R and is ignored by other clients (e.g.
+#'   pyarrow which has its own custom metadata for things like Pandas metadata).
+#'   This metadata is stored (and can be accessed with) `table$metadata$r`.
```

Comment:
Shouldn't say table here, we're in the `Schema` docs.

```suggestion
#'   This metadata is stored in `$metadata$r`.
```